admin-dev/themes/new-theme/js/pages/order/view/order-product-add.js

```diff
@@ -189,14 +191,32 @@ export default class OrderProductAdd {
     const orderId = $(event.currentTarget).data('orderId');
 
     if (invoiceId === 0) {
+      const combinationId = typeof $(':selected', this.combinationsSelect).val() === 'undefined' ? 0 : $(':selected', this.combinationsSelect).val();
+      const productPriceMatch = this.orderPricesRefresher.checkOtherProductPricesMatch(this.priceTaxIncludedInput.val(), this.productIdInput.val(), combinationId);
+
+      const modalEditPrice = new ConfirmModal({
```

Comment:
Maybe this modal should be created only in the `if (!productPriceMatch)` block, since it's not used otherwise

Reply:
This can still be optimized, a bit like you did in the edition, the modal doesn't need to be created unless the prices don't match